pkg/misc/depversion/depversion.go

```diff
@@ -159,6 +159,9 @@ var exactSvR = regexp.MustCompile(`^v?(?P<semver>(?P<major>0|[1-9]\d*)(\.(?P<min
 // check for 1.x or 1.0.x cases
 var exactSvRWithWildcard = regexp.MustCompile(`^v?(?P<semver>(?P<major>0|[1-9]\d*)(\.(?P<minor>x|0|[1-9]\d*))?(\.(?P<patch>0|x|[1-9]\d*))?(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$`)
 
+// check for ~1.x or ~1.0.x cases
+var exactSvRWithWildcardAndTidle = regexp.MustCompile(`~.*[.]x`)
+
```

Comment:
This regex looks too simplistic to catch what we want. I'd rather just test for and remove tilde if found. (below)